phoenix-core/src/main/java/org/apache/phoenix/mapreduce/index/IndexScrutinyMapper.java

```diff
@@ -228,8 +257,17 @@ private void processBatch(Context context)
                     buildTargetStatement(targetStatement);
 
             // fetch results from the target table and output invalid rows
+            if (isTest) {
```

Comment:
Usually not a good idea to put test logic that directly in production code. Injection would be cleaner, but given the M/R framework reflectively instantiates the mapper, hard to do. One option would be to have a protected hook that a test-only subclass could override that will set up the clock, and have your test use the subclass.